agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4551,13 +4561,21 @@ bool backhaul_manager::handle_backhaul_steering_request(ieee1905_1::CmduMessageR
         LOG(DEBUG) << "Sending ACK message to the originator, mid=" << std::hex << mid;
         send_cmdu_to_bus(cmdu_tx, controller_bridge_mac, bridge_info.mac);
 
-        /*
-            TODO: Need to add sending Backhaul STA Steering Response with the FAILURE error code.
-        */
+        auto response = create_backhaul_steering_response(
+            wfa_map::tlvBackhaulSteeringResponse::eResultCode::FAILURE);
```

Comment:
As mentioned earlier, in case of failure we also need to give an error code. We have the choice of BACKHAUL_STEERING_REQUEST_REJECTED_CANNOT_OPERATE_ON_CHANNEL_SPECIFIED, BACKHAUL_STEERING_REQUEST_REJECTED_TARGET_BSS_SIGNAL_NOT_SUITABLE or BACKHAUL_STEERING_REQUEST_AUTHENTICATION_OR_ASSOCIATION_REJECTED. Actually none of these are applicable here because if the roam() call failed, it just means there's a problem with the connection to wpa_supplicant. I think BACKHAUL_STEERING_REQUEST_REJECTED_TARGET_BSS_SIGNAL_NOT_SUITABLE is the closest to be accurate.

Reply:
Accepted.